legend-engine-language-pure-dsl-service/src/test/java/org/finos/legend/engine/language/pure/grammar/test/TestServiceGrammarParser.java

```diff
@@ -377,6 +377,43 @@ public void testServiceWithSingleExecution()
                 "}\n", "PARSER error at [12:9-17:3]: Field 'asserts' should be specified only once");
     }
 
+    @Test
+    public void testFaultyServiceWithOnlyQuery()
+    {
+        test("###Service\n" +
+                "Service meta::pure::myServiceSingle\n" +
+                "{\n" +
+                "  pattern: 'url/myUrl/';\n" +
+                "  documentation: 'this is just for context';\n" +
+                "  execution: Single\n" +
+                "  {\n" +
+                "    query: src: meta::transform::tests::Address[1]|$src.a->from(meta::myMapping, meta::myRuntime);\n" +
+                "    mapping: meta::myMapping;\n" +
+                "  }\n" +
+                "  test: Single\n" +
+                "  {\n" +
+                "    data: 'moreThanData';\n" +
+                "    asserts: [];\n" +
+                "  }\n" +
+                "}\n", "PARSER error at [6:14-10:3]: Field 'runtime' is required");
```

Comment:
Same as my comment above. 'runtime' is not required right ? Without 'runtime', this will be a valid inline lambda ?